Modules/System/Upgrade Tags/src/UpgradeTag.Codeunit.al

```diff
@@ -74,7 +93,7 @@ codeunit 9999 "Upgrade Tag"
     /// List of upgrade tags that should be inserted if they do not exist.
     /// </param>
     [IntegrationEvent(false, false)]
-    internal procedure OnGetPerCompanyUpgradeTags(var PerCompanyUpgradeTags: List of [Code[250]])
+    procedure OnGetPerCompanyUpgradeTags(var PerCompanyUpgradeTags: List of [Code[250]])
```

Comment:
Could you add a public method that returns the list of PerCompanyUpgradeTags to avoid exposing the event directly?
Then it will be visible that it is by design - that we want to have this within the interface. We can also control in which flows does this event get triggered.

Reply:
procedure GetPerCompanyUpgradeTags added